src/benchmarks/pmemobj_gen.cpp

```diff
@@ -542,8 +542,18 @@ pobj_direct_op(struct benchmark *bench, struct operation_info *info)
 	auto *bench_priv = (struct pobj_bench *)pmembench_get_priv(bench);
 	auto *pw = (struct pobj_worker *)info->worker->priv;
 	size_t idx = bench_priv->obj(info->index);
-	if (pmemobj_direct(pw->oids[idx]) == nullptr)
-		return -1;
+	/* Query an invalid oid:off pair to invalidate the cache. */
+	PMEMoid bad = {1, 1};
+#define OBJ_DIRECT_NITER 1024
```

Comment:
We usually put `#define`'s at the top of the file. If you want to have it in a function body, please `#undef` it right before closing parenthesis.